### DISCUSSION TITLE
Switching to Specific Tab Index

### DISCUSSION THREAD

**Discussion by OmarMoataz** [zed-industries/zed#17857] [Feature Requests]
hey folks,

I'm trying out Zed and was wondering if there's a way to bind cmd+{num} to specific tab in my window.

This is similar to [this question](https://github.com/zed-industries/zed/discussions/6903) but not quite

A similar config to that of vs code you can find here:
`
[
    {
        "key": "cmd+1",
        "command": "workbench.action.openEditorAtIndex1"
    },
    {
        "key": "cmd+2",
        "command": "workbench.action.openEditorAtIndex2"
    },
    {
        "key": "cmd+3",
        "command": "workbench.action.openEditorAtIndex3"
    },
    {
        "key": "cmd+4",
        "command": "workbench.action.openEditorAtIndex4"
    },
    {
        "key": "cmd+5",
        "command": "workbench.action.openEditorAtIndex5"
    },
    {
        "key": "cmd+6",
        "command": "workbench.action.openEditorAtIndex6"
    },
    {
        "key": "cmd+7",
        "command": "workbench.action.openEditorAtIndex7"
    },
    {
        "key": "cmd+8",
        "command": "workbench.action.openEditorAtIndex8"
    },
    {
        "key": "cmd+9",
        "command": "workbench.action.openEditorAtIndex9"
    },
    {
        "key": "cmd+l cmd+f",
        "command": "eslint.executeAutofix"
    },
    {
        "key": "cmd+0",
        "command": "workbench.action.openLastEditorInGroup"
    },
    {
        "key": "ctrl+1",
        "command": "workbench.action.focusFirstEditorGroup"
    },
    {
        "key": "ctrl+2",
        "command": "workbench.action.focusSecondEditorGroup"
    },
    {
        "key": "ctrl+3",
        "command": "workbench.action.focusThirdEditorGroup"
    }
]`

I want to be able to bind cmd+{num} to what ctrl+{num} does

OS: mac os (intel)

**Comment by thataboy** [comment]
does this work

```
  {
    "context": "Workspace",
    "bindings": {
      "cmd-1": ["workspace::ActivatePane", 0],
      "cmd-2": ["workspace::ActivatePane", 1],
      // etc
   }
```

**Reply by OmarMoataz** [reply]
Not quite,

I noticed these commands are in the default keymap and they're not working there either. Everytime I click on it, zed opens a new split tabs and focuses that. 

it looks like this:
![image](https://github.com/user-attachments/assets/e2a2aefe-e3d7-4d03-994d-a7840e2ba901)

then you can switch between the split tabs with the same commands. So I suppose my issue is being able to ActivateTab without the splitting behavior.

**Comment by thataboy** [comment]
Hmm, I have same and it's working for me. Search for cmd-1 to see if you have it defined somewhere else. Could you have an error in your keymap.json? Trash it (temporarily) to revert to default keymap and see if it works

**Reply by OmarMoataz** [reply]
Hey,

So I tried deleting the custom keymap I have and looked in the default keymap and it seems to have the commands I need: 
<img width="526" alt="image" src="https://github.com/user-attachments/assets/03c63a01-481e-43b5-8056-77ff375d1aec">

but for some reason, there is no change at all in behavior.

**Reply by OmarMoataz** [reply]
I even tried modifying cmd-1 to open the 4th active tab by mixing up the command and it made no difference. so it's clear it's not really taking into consideration the custom binding I specified. Is there a difference here between editor and workspace keymaps becuase I thought the more general would be editor so that's where I put it.

**Reply by OmarMoataz** [reply]
Okay I finally figured it out, you need to specify pane context, not editor:

<img width="612" alt="image" src="https://github.com/user-attachments/assets/6f7df1a2-6925-4164-9741-80be1edb70a6">

**Reply by nrsprs** [reply]
This works well. One note: ActivateLastItem is no longer supported, so try:
`"cmd-0": ["pane::ActivateItem", 9999],`

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
